torch/csrc/api/include/torch/data/datasets/map.h

```diff
@@ -16,21 +16,24 @@ namespace datasets {
 template <typename SourceDataset, typename AppliedTransform>
 struct MapDataset : BatchDataset<
                         MapDataset<SourceDataset, AppliedTransform>,
-                        typename AppliedTransform::OutputBatchType> {
+                        typename AppliedTransform::OutputBatchType,
+                        typename SourceDataset::IndexType> {
   using DatasetType = SourceDataset;
   using TransformType = AppliedTransform;
+  using IndexType = typename SourceDataset::IndexType;
   using OutputBatchType = typename TransformType::OutputBatchType;
 
   MapDataset(DatasetType dataset, TransformType transform)
       : dataset(std::move(dataset)), transform(std::move(transform)) {}
 
   /// Gets a batch from the source dataset and applies the transform to it,
   /// returning the result.
-  OutputBatchType get_batch(ArrayRef<size_t> indices) override {
+  OutputBatchType get_batch(IndexType indices) override {
```

Comment:
You're calling this type `Index` everywhere else. Would be good to keep it consistent (again -- it shows up in error messages).